src/main/java/com/mattermost/integration/figma/notification/service/FileNotificationService.java

```diff
@@ -1,62 +1,163 @@
+        context.setActingUser(new ActingUser());
+        context.getActingUser().setId(specificUserData.getMmUserId());
+        String channelId = messageService.createDMChannel(createDMChannelPayload(context));
+        messageService.sendDMMessage(createDMMessagePayload(channelId, context.getBotAccessToken(),
+                context.getMattermostSiteUrl(), figmaWebhookResponse));
+    }
+
+    private boolean checkIfTeamWebhookIsPresent(String teamId, String figmaToken) {
+        TeamWebhookInfoResponseDto teamWebhooks = figmaWebhookService.getTeamWebhooks(teamId, figmaToken);
+        teamWebhooks.getWebhooks().forEach(webhook -> figmaWebhookService.deleteWebhook(webhook.getId(), figmaToken));
+        return teamWebhooks.getWebhooks().stream().anyMatch(webhook -> webhook.getEventType().equals(FILE_COMMENT_EVENT_TYPE));
+    }
+
+    private DMChannelPayload createDMChannelPayload(Context context) {
+        String botAccessToken = context.getBotAccessToken();
+        String botUserId = context.getBotUserId();
+        String userId = context.getActingUser().getId();
+        String mattermostSiteUrl = context.getMattermostSiteUrl();
+        return new DMChannelPayload(userId, botUserId, botAccessToken, mattermostSiteUrl);
+    }
+
+    private DMMessagePayload createDMMessagePayload(String channelId, String botAccessToken,
+                                                    String mmSiteUrl, FigmaWebhookResponse figmaWebhookResponse) {
+        DMMessagePayload message = new DMMessagePayload();
+        message.setChannelId(channelId);
+        message.setMessage(createFileNotificationMessage(figmaWebhookResponse));
+        message.setToken(botAccessToken);
+        message.setMmSiteUrlBase(mmSiteUrl);
+        return message;
+    }
+
+    private boolean checkIfUserTeamIsPresent(InputPayload inputPayload) {
+        String userId = inputPayload.getContext().getOauth2().getUser().getUserId();
+        String mmSiteUrl = inputPayload.getContext().getMattermostSiteUrl();
+        String botAccessToken = inputPayload.getContext().getBotAccessToken();
+        Set<String> teamIds = getCurrentUserData(userId, mmSiteUrl, botAccessToken).getTeamIds();
+        return Objects.nonNull(teamIds) && teamIds.contains(inputPayload.getValues().getTeamId());
+    }
+
+    private UserDataDto getCurrentUserData(String userId, String mmSiteUrl, String botAccessToken) {
+        return (UserDataDto) jsonUtils.convertStringToObject(kvService.get(userId, mmSiteUrl,
+                botAccessToken), UserDataDto.class).get();
+    }
+
+    private String createFileNotificationMessage(FigmaWebhookResponse figmaWebhookResponse) {
+        return String.format("User %s mentioned you in a comment %s in a file %s",
+                figmaWebhookResponse.getTriggeredBy().getHandle(),
+                figmaWebhookResponse.getComment()[0].getText(),
```

Comment:
please verify , is there any chance , that message will be not only in 0 array element . 
what if i  write figma comment like: Hello dear @username , i wanna tell you one interesting story about @secondusername